source/falaise/snemo/testing/test_snemo_datamodel_sim_digi_data.cxx

```diff
@@ -58,35 +58,36 @@ void test1() {
 
   // Create and set some dummy data
   snemo::datamodel::sim_digi_data sddWrite;
-  // Calling back() on empty vector is *undefined behaviour*
-  // so must put one event into sim_digi_data instance before
-  // the loop can proceed
-  auto& eventCollection = sddWrite.grab_digi_events();
-  // Unreadable, but just a push back of a new handle-to-data.
-  // Note that sim_digi_data provides a "handle_type" typedef, but not the
-  // underlying type. We could derive that from the "handle_type",
-  // but that would almost double the number of characters...
-  eventCollection.push_back(snemo::datamodel::sim_digi_data::sim_digi_event_handle_type{
-      new snemo::datamodel::sim_digi_event_data});
-
-  for (std::size_t icalohit = 0; icalohit < 3; icalohit++) {
-    snemo::datamodel::sim_calo_digi_hit dummy;
-    // That's way, way too many leveks of nesting...
-    sddWrite.grab_digi_events().back().grab().grab_readout_data().grab_calo_digi_hits().push_back(
-        dummy);
-
-    // .... ...
-    snemo::datamodel::sim_calo_digi_hit& calo_hit =
-        sddWrite.grab_digi_events().back().grab().grab_readout_data().grab_calo_digi_hits().back();
-    calo_hit.set_hit_id(icalohit);
-    calo_hit.grab_geom_id().set_type(1234);
-    calo_hit.grab_geom_id().set_address(1, 3, 8 - icalohit, 3 + icalohit);
-    calo_hit.set_sampling_frequency(1.0 * gigahertz);
-    calo_hit.set_number_of_samples(128, 0);
-  }
+
+  auto & calo_collection    = sddWrite.grab_calo_digi_hits();
+  // auto & tracker_collection = sddWrite.grab_tracker_digi_hits();
+  // auto & trigger_collection = sddWrite.grab_trigger_digi_data();
+
+
+  for (std::size_t icalohit = 0; icalohit < 3; icalohit++)
+    {
+      snemo::datamodel::sim_digi_data::calo_digi_hit_handle_type new_handle(new snemo::datamodel::sim_calo_digi_hit);
+      snemo::datamodel::sim_calo_digi_hit & calo_hit = new_handle.grab();
+
+      calo_hit.set_hit_id(icalohit);
+      calo_hit.grab_geom_id().set_type(1234);
+      calo_hit.grab_geom_id().set_address(1, 3, 8 - icalohit, 3 + icalohit);
+      calo_hit.set_sampling_frequency(1.0 * gigahertz);
+      calo_hit.set_number_of_samples(128, 0);
+      calo_hit.tree_dump();
+
+      calo_collection.push_back(new_handle);
+    }
 
   // Write the data to bare xml file
   sddWrite.tree_dump(std::clog, "Simulated Digitized Data: ");
+
+  std::clog << "Calo collection size = " << calo_collection.size() << std::endl;
+  for (auto itcalo = calo_collection.begin(); itcalo != calo_collection.end(); itcalo++)
+    {
+      itcalo->get().tree_dump();
+    }
+
   datatools::data_writer writer("test_snemo_datamodel_sim_digi_data.xml",
                                 datatools::using_multi_archives);
   writer.store(sddWrite);
```

Comment:
This test is broken during the READ (not the store). The code compiles but, here is the error during execution : 
`[fatal:int main(int, char**):45] [void datatools::data_reader::basic_load(Data&) [with Data = snemo::datamodel::sim_digi_data]:684: input stream error]`
So I add  a new handle to the calorimeter collection and then take the reference to the calo hit in order to update it. I do it 3 times for 3 calo. Then I push back them in the calorimeter collection. The store method of the writer works because it produce the file.
The execution error happens during the `reader.read(sddReadback)`. I don't understand why the reader can not read and restore the `snemo::datamodel::sim_digi_data` object. 
Fun fact : the test works at execution if I push back only 1 handle (so with 1 calo_digi hit).

Reply:
Error found : it was because of boost::serialization on uninitialized boolean. It can store boolean even if the value is not 0 or 1 but it can not load / restore the value and it threw an exception.
In the source code of boost serialization it says : 
```
// trap usage of invalid uninitialized boolean which would
// otherwise crash on load.
```
But the error is not manage here during the store, only in the load. Maybe an 'error' is made in boost because it should manages this error during the storage.